samples/bluetooth/mesh/sensor_server/src/model_handler.c

```diff
@@ -65,21 +79,100 @@ static int chip_temp_get(struct bt_mesh_sensor_srv *srv,
 
 	return err;
 }
+/* Tolerance is based on nRF52832's range and accuracy (5/125 = 4%). */
+static const struct bt_mesh_sensor_descriptor chip_temp_descriptor = {
+	.tolerance = {
+		.negative = {
+			.val1 = 4,
+		},
+		.positive = {
+			.val1 = 4,
+		}
+	},
+	.sampling_type = BT_MESH_SENSOR_SAMPLING_INSTANTANEOUS,
+};
+
+static void chip_temp_range_get(struct bt_mesh_sensor_srv *srv, struct bt_mesh_sensor *sensor,
+				const struct bt_mesh_sensor_setting *setting,
+				struct bt_mesh_msg_ctx *ctx, struct sensor_value *rsp)
+{
+	rsp[0] = temp_range.start;
+	rsp[1] = temp_range.end;
+	printk("Temperature sensor lower limit: %u\n", rsp[0].val1);
+	printk("Temperature sensor upper limit: %u\n", rsp[1].val1);
+}
+
+static int chip_temp_range_set(struct bt_mesh_sensor_srv *srv, struct bt_mesh_sensor *sensor,
+			       const struct bt_mesh_sensor_setting *setting,
+			       struct bt_mesh_msg_ctx *ctx, const struct sensor_value *value)
+{
+	temp_range.start = value[0];
+	temp_range.end = value[1];
+	printk("Temperature sensor lower limit: %u\n", value[0].val1);
+	printk("Temperature sensor upper limit: %u\n", value[1].val1);
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		int err;
+		// TODO current method of storing is not really BT related, could improve
+		err = settings_save_one("temp/range", &temp_range, sizeof(temp_range));
+		if (err) {
+			printk("Error storing setting (%d)\n", err);
+		} else {
+			printk("Stored setting\n");
+		}
+	}
+	return 0;
+}
+
+static struct bt_mesh_sensor_setting chip_temp_setting[] = { {
+	.type = &bt_mesh_sensor_dev_op_temp_range_spec,
+	.get = chip_temp_range_get,
+	.set = chip_temp_range_set,
+} };
+
+static int chip_temp_range_settings_restore(const char *name, size_t len, settings_read_cb read_cb,
+					    void *cb_arg)
+{
+	const char *next;
+	int rc;
+
+	if (settings_name_steq(name, "range", &next) && !next) {
+		if (len != sizeof(temp_range)) {
+			return -EINVAL;
+		}
+
+		rc = read_cb(cb_arg, &temp_range, sizeof(temp_range));
+		if (rc >= 0) {
+			printk("Restored temperature range setting\n");
+			return 0;
+		}
+
+		return rc;
+	}
+
+	return -ENOENT;
+}
+
+struct settings_handler temp_range_conf = { .name = "temp",
+					    .h_set = chip_temp_range_settings_restore };
 
 static struct bt_mesh_sensor chip_temp = {
 	.type = &bt_mesh_sensor_present_dev_op_temp,
 	.get = chip_temp_get,
+	.descriptor = &chip_temp_descriptor,
+	.settings = {
+		.list = &chip_temp_setting,
+		.count = ARRAY_SIZE(chip_temp_setting),
+	},
```

Comment:
```suggestion
	.descriptor = &chip_temp_descriptor,
	.settings = {
		.list = (const struct bt_mesh_sensor_setting *)&chip_temp_setting,
		.count = ARRAY_SIZE(chip_temp_setting),
	},
```
Fixes compile warning